pkg/trace/stats/aggregation.go

```diff
@@ -99,15 +114,13 @@ func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregation
 		},
 	}
 	var peerTags []string
-	if len(peerTagKeys) > 0 && shouldCalculateStatsOnPeerTags(agg.SpanKind) {
-		for _, t := range peerTagKeys {
-			if v, ok := s.Meta[t]; ok && v != "" {
-				v = obfuscate.QuantizePeerIPAddresses(v)
-				peerTags = append(peerTags, t+":"+v)
-			}
+	for _, t := range peerTagKeysToAggregateForSpan(agg.SpanKind, s.Meta[tagBaseService], peerTagKeys) {
+		if v, ok := s.Meta[t]; ok && v != "" {
+			v = obfuscate.QuantizePeerIPAddresses(v)
+			peerTags = append(peerTags, t+":"+v)
 		}
-		agg.PeerTagsHash = peerTagsHash(peerTags)
 	}
+	agg.PeerTagsHash = peerTagsHash(peerTags)
```

Comment:
Any reason the PeerTagsHash is always attached now while previously we only calculated it and attached it when peerTags was non-empty?

Reply:
> Any reason the PeerTagsHash is always attached now while previously we only calculated it and attached it when peerTags was non-empty?

Yes, it just made the code simpler. The result is still identical, where if there are no peer tags it's zero. 

```
func peerTagsHash(tags []string) uint64 {
	if len(tags) == 0 {
		return 0
	}
```